src/App.tsx

```diff
@@ -1,50 +1,112 @@
-import React from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import { useState } from 'react';
+import cn from 'classnames';
+import React from 'react';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+'Dumplings',
+'Carrot',
+'Eggs',
+'Ice cream',
+'Apple',
+'Bread',
+'Fish',
+'Honey',
+'Jam',
+'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  none,
+  name,
+  length,
+}
+
+function prepareGoods(
+  goodsList: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goodsList];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.name:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.none);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods: string[] = prepareGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
```

Comment:
You can rewrite like this
```suggestion
          className={cn('button is-info', {
```